### ISSUE TITLE
viewmail.php does not display foreign language subjects correctly.

### ISSUE THREAD

**Issue by pdwalker** [mailwatch/MailWatch#773]
Subject decoding is correct in status.php 
![status php](https://cloud.githubusercontent.com/assets/93068/25690817/c1c5abfa-30c7-11e7-9c8c-f4c61e01c70e.png)

and detail.php, 
![detail php](https://cloud.githubusercontent.com/assets/93068/25690821/c94f7644-30c7-11e7-9e6b-621671c468a2.png)

but not in viewmail.php
![viewmail php](https://cloud.githubusercontent.com/assets/93068/25690828/cf3a2a36-30c7-11e7-8ac4-aa7667017067.png)

The actual subject header is:
```
Subject: =?UTF-8?B?5aaC5L2V5YeG56Gu6K+G5Yir5Ye66auY5r2c6LSo55qE5Lq65omN77yf6auY?=
 =?UTF-8?B?5r2c5Lq65omN6aG555uu55qE5bi46KeB5oyR5oiY5pyJ5ZOq5Lqb77yfLS0t?=
 =?UTF-8?B?LeWJjeaZuuedv+WSqOivokRESeWkp+S4reWNjuWMukdN5YiY5Lyf?=
 =?UTF-8?B?5biI?=
```

**Comment by stefaweb** [comment]
Could you check if the text is well encoded in database?

**Comment by pdwalker** [comment]
```
MariaDB [mailscanner]> select subject from maillog where id = '2D49418005D.A7ADC';
+--------------------------------------------------------------------------------------------------------------------------------------+
| subject                                                                                                                              |
+--------------------------------------------------------------------------------------------------------------------------------------+
| 如何准确识别出高潜质的人才？高潜人才项目的常见挑战有哪些？----前智睿咨询DDI大中华区GM刘伟师                                          |
+--------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

Looks like the encoding in the DB is good to me.

Is there anything else I can check?

**Comment by stefaweb** [comment]
For me, the problem can only come from your PHP version/config.

**Comment by pdwalker** [comment]
In https://github.com/mailwatch/MailWatch/issues/742 I was able to verify the encoding of the database and data stored in the fields as apparently being correct.  

Checking the test message that I sent in #742 shows the same thing - correct in status.php and detail.php, 
![screen shot 2017-05-04 at 16 16 2](https://cloud.githubusercontent.com/assets/93068/25695388/7af41522-30e5-11e7-971a-39106c4b6f44.png)

but jibberish in viewmail.php
![screen shot 2017-05-04 at 16 16](https://cloud.githubusercontent.com/assets/93068/25695325/3742382c-30e5-11e7-90ce-ce3c112fa923.png)

**Comment by pdwalker** [comment]
Possibly, as I'm using PHP 5.3.3.

I just find it unusual that it works correctly in one place, but not another.

**Comment by stefaweb** [comment]
> I just find it unusual that it works correctly in one place, but not another.
The code is may be not the same to print Subject.

**Comment by pdwalker** [comment]
I've found what is causing the problem... one moment while I verify the fix

**Comment by pdwalker** [comment]
in [viewmail.php](https://github.com/mailwatch/MailWatch/blame/master/mailscanner/viewmail.php), there is a call to "htmlentities($v)" on line 139 of the current code.  This change was made because of issue #123 

If I comment that part out, the subject displays correctly.

I don't know why the fields in viewmail.php are subject to the [htmlentities](http://php.net/manual/en/function.htmlentities.php) function while they are not in display.php and status.php.

Ah... from the php documentation:
> If omitted, the default value of the encoding varies depending on the PHP version in use. In PHP 5.6 and later, the default_charset configuration option is used as the default value. PHP 5.4 and 5.5 will use UTF-8 as the default. Earlier versions of PHP use ISO-8859-1.

So yes, technically it is because I'm using an older version of PHP.

**Comment by pdwalker** [comment]
ok, the fix for me using PHP 5.3  is to change:
`$v = htmlentities($v);`
to
`$v = htmlentities($v,ENT_NOQUOTES,ini_get("default_charset"));`
and then make sure my "default_charset" php.ini option is set to "utf-8"

So for an actual "fix", then If mailscanner doesn't specify it, it should recommend at least PHP 5.4 for full working multilanguage support.

Thanks for the pointer.
